nyx/nyx_gui/frontend/gui_tools_partition_manager.c

```diff
@@ -1720,12 +1748,74 @@ static lv_res_t _create_mbox_partitioning_next(lv_obj_t *btn)
 
 	lv_obj_align(mbox, NULL, LV_ALIGN_CENTER, 0, 0);
 	lv_obj_set_top(mbox, true);
+	manual_system_maintenance(true);
 
 	free(txt_buf);
 
 	return LV_RES_OK;
 }
 
+static lv_res_t _create_mbox_partitioning_android(lv_obj_t *btns, const char *txt)
+{
+	int btn_idx = lv_btnm_get_pressed(btns);
+
+	mbox_action(btns, txt);
+
+	switch (btn_idx)
+	{
+		case 0:
+			part_info.and_dynamic = true;
+			_create_mbox_partitioning_warn(NULL);
+			break;
+		case 1:
+			part_info.and_dynamic = false;
+			_create_mbox_partitioning_warn(NULL);
+			break;
+		default:
+			break;
+	}
+
+	return LV_RES_OK;
```

Comment:
Ah, I see. I didn't understand the significance of `mbox_action`--I'll remove the usage. The default case will have to then have `mbox_action` and return INV there though, right? Because it is a cancel button? Or am I not understanding?

Reply:
The opposite. mbox_action is needed for any button that deletes the pop up. when it gets deleted you invalidate.
Padding buttons can also proc the action if they don't have correct flags, so that's why you normally guard against them.
(No mbox_action and OK).

Cancel also should not exist. Didn't see it and I guess that's why there's a default.

So without a cancel:
```c
static lv_res_t _create_mbox_partitioning_android(lv_obj_t *btns, const char *txt)
{
	int btn_idx = lv_btnm_get_pressed(btns);

	mbox_action(btns, txt);

	part_info.and_dynamic = !btn_idx;
	_create_mbox_partitioning_warn(NULL);

	return LV_RES_INV;
}
```
The above assumes 2 buttons and `android13, android10-11` order.
(if padding is added the btn index changes)

Reply:
I was only thinking there would be a cancel because it is the first in the series, but that makes sense. I'll use that.